src/commands.rs

```diff
@@ -36,13 +37,79 @@ pub enum EnvAction {
     Remove,
 }
 
+/// Define what strategy to use when choosing privilege scalation programs.
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ElevationStrategy {
+    Auto,
+    Prefer(OsString),
+    Force(&'static str),
+}
+
+impl ElevationStrategy {
+    pub fn resolve(&self) -> Result<OsString> {
+        match self {
+            ElevationStrategy::Auto => Self::choice(),
+            ElevationStrategy::Prefer(program) => which(program)
+                .map(std::path::PathBuf::into_os_string)
+                .or_else(|_| {
+                    let auto = Self::choice()?;
+                    warn!(
+                        "{} not found. Using {} instead",
+                        program.to_string_lossy(),
+                        auto.to_string_lossy()
+                    );
+                    Ok(auto)
+                }),
+            ElevationStrategy::Force(program) => Ok(program.into()),
+        }
+    }
+
+    /// Gets a path to a previlege elevation program based on what is available in the system.
+    ///
+    /// This funtion checks for the existence of common privilege elevation program names in
+    /// the `PATH` using the `which` crate and returns a Ok result with the `OsString` of the
+    /// path to the binary. In the case none of the checked programs are found a Err result is
+    /// returned.
+    ///
+    /// The search is done in this order:
+    ///
+    /// 1. `doas`
+    /// 1. `sudo`
+    /// 1. `run0`
+    /// 1. `pkexec`
```

Comment:
```suggestion
    /// 1. `doas`
    /// 2. `sudo`
    /// 3. `run0`
    /// 4. `pkexec`
```

Formatter issue?

Reply:
Oh that's neat, didn't know Rustdoc format supported it.

Reply:
One of Markdown's strengths is that it's easy to read as a human even without formatting. Auto-renumbering kind of hurts that and I've found more cases where I wish it didn't happen than did. 

In this case, even if it can be auto-renumbered, I'd still say to number them manually so that it's easier to read when looking at the raw source code rather than just the docs. (Especially since this is an executable rather than a library, so the source code would likely be read more than the docs.)

Reply:
I really don't see how in any way this is hurting readibility. It isn't like the numbers are being referenced in any sort of way. Which number that the item in the ordered list doesn't really effect how you interpret this at all. It could just as well be an unordered list, but an ordered list was used to communicate precedence.